command/node_drain.go

```diff
@@ -68,8 +68,47 @@ func (c *NodeDrainCommand) Run(args []string) int {
 		return 1
 	}
 
+	// Check if node exists
+	node, _, err := client.Nodes().Info(nodeID, nil)
+	if err != nil {
+		// Exact lookup failed, try with prefix based search
+		nodes, _, err := client.Nodes().PrefixList(nodeID)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error toggling drain mode: %s", err))
+			return 1
+		}
+		// Return error if no nodes are found
+		if len(nodes) == 0 {
+			c.Ui.Error(fmt.Sprintf("No node(s) with prefix or id %q found", nodeID))
+			return 1
+		}
+		if len(nodes) > 1 {
+			// Format the nodes list that matches the prefix so that the user
+			// can create a more specific request
+			out := make([]string, len(nodes)+1)
```

Comment:
I'm not sure how you feel about the way I put it in the code with two newlines. I didn't see a nicer way to display the help (perhaps by expanding the array and adding the line(s) in there). I really think there should be a line between the disambiguation message and the list.